README.md

```diff
@@ -32,47 +32,70 @@ This must only be done once per component.
 An empty component looks as follows:
 
 ```Ada
-with Cai.Types;
-with Cai.Component;
+with Componolit.Gneiss.Types;
+with Componolit.Gneiss.Component;
 
 package Component is
 
-   procedure Initialize (Cap : Cai.Types.Capability);
+   procedure Construct (Cap : Componolit.Gneiss.Types.Capability);
+   procedure Destruct;
    
-   package My_Component is new Cai.Component (Initialize);
+   package Main is new Componolit.Gneiss.Component (Construct, Destruct);
    
 end Component;
 ```
 
-The procedure `Initialize` is the first called procedure of the component (except of elaboration code) and receives a valid system capability. This capability is required to initialize clients or register servers.
+The procedure `Construct` is the first called procedure of the component (except of elaboration code) and receives a valid system capability. This capability is required to initialize clients or register servers.
+The procedure `Destruct` is called when the platform decides to stop the component and allows to finalize component state.
+As a convention the components main package is always called `Component` and it must contain an instance of `Componolit.Gneiss.Component` that is named `Main`.
 
-### Using a client
-
-Clients are used like regular libraries. The simples client is the `Log` client that provides standard logging facilities.
+The simplest interfaces are used like regular libraries. An example is the `Log` client that provides standard logging facilities.
 A hello world with the component described above looks as follows
 
 ```Ada
-with Cai.Log;
-with Cai.Log.Client;
+with Componolit.Gneiss.Log;
+with Componolit.Gneiss.Log.Client;
 
 package body Component is
 
-   procedure Initialize (Cap : Cai.Types.Capability)
+   Log : Componolit.Gneiss.Log.Client_Session;
+
+   procedure Construct (Cap : Componolit.Gneiss.Types.Capability)
    is
-      Log : Cai.Log.Client_Session;
    begin
-      Cai.Log.Client.Initialize (Log, Cap, "Hello_World");
-      Cai.Log.Client.Info (Log, "Hello World!");
-      Cai.Log.Client.Warning (Log, "Hello World!");
-      Cai.Log.Client.Error (Log, "Hello World!");
-      Cai.Log.Client.Finalize (Log);
-   end Initialize;
+      if not Componolit.Gneiss.Log.Initialized (Log) then
+         Componolit.Gneiss.Log.Client.Initialize (Log, Cap, "Hello_World");
```

Comment:
What is "Hello_World" good for, what do I have to use here?

Reply:
Addressed by https://github.com/Componolit/gneiss/pull/92/commits/db86420560310e1980fe1d0a46a73f93c1080751.